### DISCUSSION TITLE
Choose media type before uploading media through content section

### DISCUSSION THREAD

**Discussion by sigrundis** [umbraco/Umbraco-CMS#10602] [Features and ideas]
### Umbraco Version
8.14

### Description
I love the fact that with the Media Picker 3 you can limit data/document types to specific media types. However, if a content editor wants to upload new media while editing the content, they cannot choose the type of media uploaded. 

### Example
I created a media type for SVG files only and used the editor for a field in a document type. However, if content editors want to upload a new SVG file to use for the field through the content editor (while editing the content) they cannot select the media type of the file. The file is set to a default type (in this case media type: file), and because the field is limited to media type: SVG, they cannot use the file they just uploaded. They need to upload new files through the media section to be able to select the correct media type.

### Steps to reproduce
1. Create new media type -> add property -> select File Upload -> Edit it by clicking on it -> Name it Upload SVG (or something similar) -> under Configuration, set the accepted file extensions to svg -> make the SVG upload field mandatory
![image01v2](https://user-images.githubusercontent.com/35728588/124567651-7fe83980-de44-11eb-8abd-6770192d5bd6.png)
![image02](https://user-images.githubusercontent.com/35728588/124567378-3b5c9e00-de44-11eb-80ad-5da8f2388841.png)
![image03](https://user-images.githubusercontent.com/35728588/124567507-5af3c680-de44-11eb-9858-7d731908fc14.png)

2. Create new Data Type -> Select Media Type under Property Editor -> Select the SVG media type under Accepted types
![image](https://user-images.githubusercontent.com/35728588/124559217-c5eccf80-de3b-11eb-8903-c409b55dd9c9.png)

3. Open the Document Type that should use the svg file field -> add property -> select the SVG Media Data Type as editor 
![image04](https://user-images.githubusercontent.com/35728588/124567808-9e4e3500-de44-11eb-93ae-34ee0e10c172.png)

4. Create Content using the Document Type that includes the svg media upload field -> Upload new SVG file -> By default, it gets uploaded as media type: file. Because the field is restricted to using media type: SVG only, the newly uploaded file cannot be used.
![image](https://user-images.githubusercontent.com/35728588/124559748-53c8ba80-de3c-11eb-8e67-7e821ddb0feb.png)
![image05](https://user-images.githubusercontent.com/35728588/124567974-c047b780-de44-11eb-8a7a-6748e212cf23.png)
![image06](https://user-images.githubusercontent.com/35728588/124568115-e3726700-de44-11eb-8f91-ea224a1ce304.png)
![image](https://user-images.githubusercontent.com/35728588/124568335-16b4f600-de45-11eb-9093-74c94edf7aa1.png)

### Expected result
It would be nice if the content editor could select the media type of the files uploaded from the content section.

**Comment by nul800sebastiaan** [comment]
Thanks @sigrundis - this is due to SVG by default not being set as an image file type. In order to preserve backwards compatibility, we currently can't change that (people expecting SVG to be created as a `File` will suddenly get different behavior.

This seems like a good one to fix for v9, adding svg here and mark it as breaking change: https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Core/Configuration/UmbracoSettings/ContentImagingElement.cs#L17

In the mean time, you [can add `<imaging>` to your `umbracoSettings.config`](https://our.umbraco.com/Documentation/Reference/Config/umbracoSettings/#imaging) - I think this will do the trick, add it within the `<content>` node:

```xml
<imaging>
    <imageFileTypes>jpeg,jpg,gif,bmp,png,tiff,tif,svg</imageFileTypes>
</imaging>
```

I'll put this as up for grabs to fix in v9, accepting svg as an image by default.

**Reply by nul800sebastiaan** [reply]
https://github.com/umbraco/Umbraco-CMS/issues/10604

**Reply by sigrundis** [reply]
Hmm... sounds like I was not clear enough 😅  

The feature I'm requesting is that when uploading media through the content section - I want to be able to select a media type for it - like you can do when you upload it through the media section:
![image](https://user-images.githubusercontent.com/35728588/124611356-a7a1c680-de71-11eb-83e9-cc6af2b18ad9.png)

In the example described above, I don't want it to get the media type: image or file, I want it to be uploaded as my custom media type: SVG - since the field has a validation that the media used must have the type: SVG.

AND/OR 
Be able to choose the default media type for media uploaded through the content section for a specific field. 

The reason:
Improve editor experience for the content editors. When they add content for a media field, they usually do so in the content section while updating the content. It could be annoying for them needing to close what they are doing to add the media type through the media section since they get a validation error if they upload it through the content section - because they can't make sure the file is uploaded as the correct media type for the field. 

To put things to perspective, I'm using your suggestion here:
https://github.com/umbraco/Umbraco-CMS/issues/4921#issuecomment-850350891

But yeah, I want to be able to add media through the content section without getting a validation error because I can't make sure the file has the correct type if I do it from there. 

Hope I managed to explain it properly this time 😅 Or am I misunderstanding something, is it maybe possible? 😊

**Reply by nul800sebastiaan** [reply]
Thanks again for the feedback @sigrundis. I was clearly tired yesterday and didn't spot the problem that I should have seen in your first post. This is a bit silly buy Umbraco needs you to create the alias of your upload field specifically to be `umbracoFile`, otherwise it doesn't really know where to look for the file 🙈  This is a weird quirk that comes from far in the depths of Umbraco history (probably that code for it was created about 10-15 years ago). 

So when I  do that, I don't need to update the `imageFileTypes`, Umbraco just understand that your SVG media type is the only one that will accept and svg file and doesn't give you a choice, it just does the right thing. 


https://user-images.githubusercontent.com/304656/124726735-4ed43b80-df0e-11eb-94bd-24f43a7add2e.mp4

If you leave the svg in `imageFileTypes` it actually DOES ask you while media type you want to create because now it is possible that it's an `Image` mediatype as well, so I would recommend to remove it from the `imageFileTypes` list so you don't need to choose, saves you a click. 🙂
